ga4gh/protocol.py

```diff
@@ -96,149 +175,28 @@ def isFull(self):
         The buffer is full if either (1) the number of items in the value
         list is >= pageSize or (2) the total length of the serialised
         elements in the page is >= maxResponseLength.
+
+        If page_size or max_response_length were not set in the request
+        then they're not checked.
         """
         return (
-            self._numElements >= self._pageSize or
-            self._valueListBuffer.tell() >= self._maxResponseLength)
+            (self._pageSize > 0 and self._numElements >= self._pageSize) or
+            (self._maxResponseLength > 0 and
```

Comment:
Why are we checking for ``self._maxResponseLength > 0`` here? It would be more readable if we had a method ``getResponseLength`` rather than running the toJson inline. (Converting to JSON each time is fine for now, but this will be a major perf bottleneck, and so it's good to abstract it out a bit)